### DISCUSSION TITLE
Add Hibernation into Cosmos (suggestion

### DISCUSSION THREAD

**Discussion by Samdasti** [CosmosOS/Cosmos#2915] [Ideas]
Suggested function to add to Cosmos userkit and devkit: Cosmos.System.Power.Hibernate(); - Hibernates the Cosmos OS Project.

Also, i am currently using the user kit, also, i want to add a Cosmos.System.Power.HibFileDirectory(<insert-path-here>);, to specify the file path of the hibernation file, this must be added before Cosmos.System.Power.Hibernate();, the hibernation file is created when the vm is hibernated.

**Comment by valentinbreiz** [comment]
Hello, this is a good suggestion but needs a good ACPI implementation. ACPI is barely supported in Cosmos but a work is in progress to port [lai](https://github.com/managarm/lai) in Cosmos: [CosmosAML](https://github.com/MishaProductions/AMLParser).

**Reply by zarlo** [reply]
we may also want paging too

**Reply by Samdasti** [reply]
But Cosmos still supports shutdown and restart, but not hibernate.
